### PR TITLE
Add more grade task

### PR DESCRIPTION
**Pull request by murugan-r**
- Pass token as spaced argument

### CODE REVIEW

**Comment by murugan-r** [comment]
@raghubetina Could you please take a look.

File: lib/tasks/grade.rake [hunk 1]
```diff
@@ -1,18 +1,43 @@
...
-  config = YAML.load_file(config_file_name)
-  git_url = `git config --get remote.origin.url`.chomp
-  username = git_url.split(':')[1].split('/')[0]
-  reponame =  git_url.split(':')[1].split('/')[1].sub(".git", "")
-  sha = `git rev-parse --verify HEAD`.chomp
-  if token.present?
-    GradeRunner::Runner.new(config['project_token'], config['submission_url'], token, rspec_output_json, username, reponame, sha, 'manual').process
-  else
-    puts "We couldn't find your access token, so we couldn't record your grade. Please click on the assignment link again and run the rails grade ...  command shown there."
+require 'optparse'
```
Comment (raghubetina) [inline lib/tasks/grade.rake]: Prefer double-quotes (we should make a Trello card to add [thoughtbot rubocop](https://github.com/thoughtbot/guides/blob/master/style/ruby/.rubocop.yml) and CodeClimate to this project).

File: lib/tasks/grade.rake [hunk 2]
```diff
@@ -1,18 +1,43 @@
...
+
+desc "Grade"
```
Comment (raghubetina) [inline lib/tasks/grade.rake]: Let's make this `desc "Alias for \"grade:next\"."`

File: lib/tasks/grade.rake [hunk 3]
```diff
@@ -1,18 +1,43 @@
...
+task grade: "grade:next" do
+end
+
+namespace :grade do
+  desc "Grade All"
```
Comment (raghubetina) [inline lib/tasks/grade.rake]: Let's make this `desc "Run all tests and submit a build report."`.

File: lib/tasks/grade.rake [hunk 4]
```diff
@@ -1,18 +1,43 @@
...
+    sha = `git rev-parse --verify HEAD`.chomp
+
+    if token.present?
+      GradeRunner::Runner.new(config['project_token'], config['submission_url'], token, rspec_output_json, username, reponame, sha, 'manual').process
+    else
+      puts "We couldn't find your access token, so we couldn't record your grade. Please click on the assignment link again and run the rails grade ...  command shown there."
+    end
+  end
+
+  desc "Grade Next"
```
Comment (raghubetina) [inline lib/tasks/grade.rake]: Let's make this `desc "Run only the next failing test."`

File: lib/tasks/grade.rake [hunk 5]
```diff
@@ -1,18 +1,43 @@
...
+  task next: :environment do
+    path = Rails.root + "examples.txt"
```
Comment (raghubetina) [inline lib/tasks/grade.rake]: Conventionally should this use `Rails.root.join`?

File: lib/tasks/grade.rake [hunk 6]
```diff
@@ -1,18 +1,43 @@
...
+task grade: "grade:next" do
+end
+
+namespace :grade do
+  desc "Grade All"
+  task all: :environment do
+    ARGV.each { |a| task a.to_sym do ; end }
+    token = ARGV[1]
+    
+    path = Rails.root + "/tmp/output" + "#{SecureRandom.hex}.json"
```
Comment (raghubetina) [inline lib/tasks/grade.rake]: Conventionally should this use `Rails.root.join`?

File: lib/tasks/grade.rake [hunk 7]
```diff
@@ -1,18 +1,43 @@
...
+    config_file_name = Rails.root.join(".firstdraft_project.yml")
+    config = YAML.load_file(config_file_name)
+
+    git_url = `git config --get remote.origin.url`.chomp
+    username = git_url.split(':')[1].split('/')[0]
+    reponame =  git_url.split(':')[1].split('/')[1].sub(".git", "")
+    sha = `git rev-parse --verify HEAD`.chomp
+
+    if token.present?
+      GradeRunner::Runner.new(config['project_token'], config['submission_url'], token, rspec_output_json, username, reponame, sha, 'manual').process
```
Comment (raghubetina) [inline lib/tasks/grade.rake]: Prefer double-quotes

File: lib/tasks/grade.rake [hunk 8]
```diff
@@ -1,18 +1,43 @@
...
+    else
+      puts "We couldn't find your access token, so we couldn't record your grade. Please click on the assignment link again and run the rails grade ...  command shown there."
```
Comment (raghubetina) [inline lib/tasks/grade.rake]: Can we cache the access token after the first run in a hidden & gitignored file the way that the original `rails grade` did?

File: lib/tasks/grade.rake [hunk 9]
```diff
@@ -1,18 +1,43 @@
...
+  desc "Grade All"
+  task all: :environment do
+    ARGV.each { |a| task a.to_sym do ; end }
+    token = ARGV[1]
+    
+    path = Rails.root + "/tmp/output" + "#{SecureRandom.hex}.json"
+    `RAILS_ENV=test bundle exec rspec --order default --format JsonOutputFormatter --out #{path}`
+
+    rspec_output_json = JSON.parse(File.read(path))
+    config_file_name = Rails.root.join(".firstdraft_project.yml")
```
Comment (raghubetina) [inline lib/tasks/grade.rake]: Can we change this to assume that the URL is `https://grades.firstdraft.com/builds` if `.firstdraft_project.yml` is not present? We only need to specify during development, to point it at staging or local; for real projects I would prefer not to include `.firstdraft_project.yml` at all.

File: lib/tasks/grade.rake [hunk 10]
```diff
@@ -1,18 +1,43 @@
...
+    else
+      puts "We couldn't find your access token, so we couldn't record your grade. Please click on the assignment link again and run the rails grade ...  command shown there."
+    end
+  end
+
+  desc "Grade Next"
+  task next: :environment do
+    path = Rails.root + "examples.txt"
+    if File.exist?(path)
+      puts `RAILS_ENV=test bundle exec rspec --next-failure`
```
Comment (raghubetina) [inline lib/tasks/grade.rake]: Can we check if a hint is present on the failing test and if so output the hint also?

**Pull request merged**